docs/design/README.md

```diff
@@ -387,6 +416,69 @@ Directory descriptions:
 * packages: Where the RPM packages are defined.
 * variants: Where Bottlerocket image builds are defined.
 
+#### Project Variations
+
+Twoliter projects can serve one or both of two purposes.
+1) A Twoliter project may be used to create variant images.
+2) A Twoliter project may be used to create kits.
+
+Each of the above purposes may be achieved with or without depending on external kits[^nokits].
+
+[^nokits]: The use case of creating a variant without depending on external kits may seem remote.
+    But at least initially, this is what the Bottlerocket monorepo will be doing.
+
+When a user has no intention of creating a kit, it should not be necessary to define a kit in the kits directory.
+When a user has no intention of creating a variant image, it should not be necessary to define a variant in the variants directory.
+
+To further complicate matters, some users may have projects with which have started with the [alpha](#twoliter-alpha) version of Twoliter.
+For the time being, if we assume backward compatibility with these projects, then they introduce another use case.
```

Comment:
I'm open to breaking compatibility here, possibly with a fix-up script or tool that can help with the migration.